crates/api_common/src/utils.rs

```diff
@@ -743,36 +769,111 @@ pub async fn remove_user_data(
     .await?;
   }
 
-  // Comments
-  Comment::update_removed_for_creator(pool, banned_person_id, true).await?;
-
   Ok(())
 }
 
 /// We can't restore their images, but we can unremove their posts and comments
 pub async fn restore_user_data(
+  mod_person_id: PersonId,
   banned_person_id: PersonId,
+  reason: &Option<String>,
   context: &LemmyContext,
 ) -> LemmyResult<()> {
   let pool = &mut context.pool();
+  let removed = false;
 
   // Posts
-  Post::update_removed_for_creator(pool, banned_person_id, None, false).await?;
+  let restored_posts =
+    Post::update_removed_for_creator(pool, banned_person_id, None, removed).await?;
+  create_modlog_entries_for_removed_or_restored_posts(
+    pool,
+    mod_person_id,
+    restored_posts.iter().map(|r| r.id).collect(),
+    removed,
+    reason,
+  )
+  .await?;
 
   // Comments
-  Comment::update_removed_for_creator(pool, banned_person_id, false).await?;
+  let restored_comments =
+    Comment::update_removed_for_creator(pool, banned_person_id, removed).await?;
+  create_modlog_entries_for_removed_or_restored_comments(
+    pool,
+    mod_person_id,
+    restored_comments.iter().map(|r| r.id).collect(),
+    removed,
+    reason,
+  )
+  .await?;
+
+  Ok(())
+}
+
+async fn create_modlog_entries_for_removed_or_restored_posts(
+  pool: &mut DbPool<'_>,
+  mod_person_id: PersonId,
+  post_ids: Vec<PostId>,
+  new_removed: bool,
+  reason: &Option<String>,
```

Comment:
[Same as here.](https://github.com/LemmyNet/lemmy/pull/5050/files#r1774027181)

Reply:
The modlog one was easy to do in a reversible way.